contracts/Gateway/NativeTokenGateway.sol

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.20;
+
+import { Ownable2Step } from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import { SafeERC20, IERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { ReentrancyGuard } from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+import { IWrappedNative } from "./Interfaces/IWrappedNative.sol";
+import { INativeTokenGateway } from "./INativeTokenGateway.sol";
+import { IVToken } from "./Interfaces/IVToken.sol";
+
+/**
+ * @title NativeTokenGateway
+ * @author Venus
+ * @notice NativeTokenGateway contract facilitates interactions with a vToken market for native tokens (Native or wNativeToken)
+ */
+contract NativeTokenGateway is INativeTokenGateway, Ownable2Step, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @notice Address of wrapped native token contract
+     */
+    IWrappedNative public immutable wNativeToken;
+
+    /**
+     * @notice Address of wrapped native token market
+     */
+    IVToken public immutable vWNativeToken;
+
+    /**
+     * @notice Constructor for NativeTokenGateway
+     * @param wrappedNativeToken Address of wrapped native token contract
+     * @param vWrappedNativeToken Address of wrapped native token market
+     */
+    constructor(IWrappedNative wrappedNativeToken, IVToken vWrappedNativeToken) {
+        ensureNonzeroAddress(address(wrappedNativeToken));
+        ensureNonzeroAddress(address(vWrappedNativeToken));
+
+        wNativeToken = wrappedNativeToken;
+        vWNativeToken = vWrappedNativeToken;
+    }
+
+    /**
+     * @notice To receive Native when msg.data is empty
+     */
+    receive() external payable {}
+
+    /**
+     * @notice To receive Native when msg.data is not empty
+     */
+    fallback() external payable {}
```

Comment:
We need the `receive` function to receive eth from the user, we can drop the `fallback` function.
WDYT?

Reply:
We need the receive function as we are sending the native tokens to the user in `_safeTransferNativeTokens()` for which it is necessary otherwise the tx fails.

Reply:
In that case, we could remove the `fallback()` function. If we don't need it, I would remove it.